Improve support for ignoring sub directories (#1517)
ocaml/dune: doc/dune-files.rst

ignored_subdirs
---------------

The ``ignored_subdirs`` stanza allows to tell Dune to ignore one or more
sub-directories. The syntax is based on dune's predicate language and allows the
user the following operations:

> - Special values. ``:all`` refers to all sub directories.

Reviewer: Do we still need this? It's basically the same as `*`

Author: `:all` is quite a bit more efficient. Is this optimization important to keep?

Author: Actually, I think `:all` is not gonna make so much sense if we allow for globs outside the current dir. For example, if we want a recursive glob, we'll have `**`. In this situation, `:all` is really `**`.

Reviewer: If it's just an optimisation, then we could effectively optimise all `*` globs, so yh I don't think it's worth keeping `:all`

Author: I got rid of it. The optimization won't even buy much in the common case where the value will be `:standard`.

- Set operations. Differences are expressed with backslash: ``:all \ bar``,
  unions are just concatenation.

- Sets can be defined using globs.

Examples:

.. code:: scheme

   (ignored_subdirs :all) ;; ignore all directories

>    (ignored_subdirs (:all \ ocaml)) ;; ignore all directories except ocaml

Reviewer: We don't need the parentheses here, right?

ignored_subdirs
---------------

The ``ignored_subdirs`` stanza allows to tell Dune to ignore one or more
sub-directories. The syntax is based on dune's predicate language and allows the
user the following operations:

- The special value ``:standard`` which refers to the default set of ignored

>   directories. These are the directories that start with any of ``[._#]``.

Reviewer: ```suggestion
  directories. These are the directories that start with ``.`` or ``_``.
```

Otherwise it might not be clear that this is a glob. BTW, why `#`? We didn't use to ignore directories starting with `#`.
